### PR TITLE
Add endpoint to get all user reserved blocks

### PR DESCRIPTION
**Pull request by justinkonecny**
- Add endpoint `GET /api/v1/protected/blocks/reserved` to return all blocks that are `assigned_to` the current user and have a status of `RESERVED`
  - Including query param `done=true` will also return all blocks withs status `DONE`
  - Returns a list of `block.fid`

### CODE REVIEW

**Review by jackblanc — DISMISSED** [review]
Thanks!

File: api/src/main/java/com/codeforcommunity/rest/subrouter/BlocksRouter.java [hunk 1]
```diff
@@ -93,4 +101,15 @@ private void handleResetRoute(RoutingContext ctx) {
 
     end(ctx.response(), 200, JsonObject.mapFrom(response).encode());
   }
+
+  private void handleGetReserved(RoutingContext ctx) {
+    JWTData userData = ctx.get("jwt_data");
+
+    boolean includeDone =
```
Comment (liammoyn) [inline api/src/main/java/com/codeforcommunity/rest/subrouter/BlocksRouter.java]: Use a method in RestFunctions for this that can do null checking properly (if there's not one for query params then create it)

**Review by liammoyn — CHANGES_REQUESTED** [review]
Please add a specification of this route to the api-spec for sftt in the dev-spikes repo

File: api/src/main/java/com/codeforcommunity/rest/subrouter/BlocksRouter.java (hunk 1 above)
Comment (justinkonecny) [reply api/src/main/java/com/codeforcommunity/rest/subrouter/BlocksRouter.java]: Done

**Comment by justinkonecny** [comment]
> Please add a specification of this route to the api-spec for sftt in the dev-spikes repo

Done

**Review by liammoyn — APPROVED** [review]

**Pull request merged**